securedrop/i18n.py

```diff
@@ -165,11 +172,17 @@ def map_locale_display_names(config: SDConfig) -> None:
     """
     language_locale_counts = collections.defaultdict(int)  # type: Dict[str, int]
     for l in sorted(config.SUPPORTED_LOCALES):
+        if Locale.parse(l) not in USABLE_LOCALES:
+            continue
+
         locale = RequestLocaleInfo(l)
         language_locale_counts[locale.language] += 1
 
     locale_map = collections.OrderedDict()
     for l in sorted(config.SUPPORTED_LOCALES):
+        if Locale.parse(l) not in USABLE_LOCALES:
```

Comment:
Seems weird to do this twice. Can't we combine the two for loops pretty trivially?

```python3
    for l in sorted(config.SUPPORTED_LOCALES):
        if Locale.parse(l) not in USABLE_LOCALES:
            continue

        locale = RequestLocaleInfo(l)
        language_locale_counts[locale.language] += 1
        if language_locale_counts[locale.language] > 1:
            locale.use_display_name = True
            locale_map[str(locale)] = locale
```

Or a set based approach:
```python3
    seen: Set[str] = set()
    locale_map = collections.OrderedDict()
    for l in sorted(config.SUPPORTED_LOCALES):
        if Locale.parse(l) not in USABLE_LOCALES:
            continue

        locale = RequestLocaleInfo(l)
        if locale.language in seen:
            # This is the second time it's appeared
            locale.use_display_name = True
            locale_map[str(locale)] = locale
        else:
            seen.add(locale.language)
```

Reply:
Absolutely; I was going for the minimal intervention here.  :-) Refactored (with corrected logic) per your second approach in bc5e3661babc4ff8af7fb8fe18b542daeef1b984; test plan updated accordingly.